pkg/extension/sumologicextension/credentials.go

```diff
@@ -0,0 +1,182 @@
+	"go.uber.org/zap"
+)
+
+// CredsGetter is an interface to get collector authentication data
+type CredsGetter interface {
+	CheckCollectorCredentials() bool
+	GetStoredCredentials() (api.OpenRegisterResponsePayload, error)
+	RegisterCollector(ctx context.Context) (api.OpenRegisterResponsePayload, error)
+}
+
+// credsGetter is a common structure for both types of getting collector credentials
+// stored in file and registration new collector.
+type credsGetter struct {
+	conf   *Config
+	logger *zap.Logger
+}
+
+// CheckCollectorCredentials checks if collector credentials can be found in path
+// configured in the config.
+func (cr credsGetter) CheckCollectorCredentials() bool {
+	filenameHash, err := hash(cr.conf.CollectorName)
+	if err != nil {
+		return false
+	}
+	path := path.Join(cr.conf.CollectorCredentialsPath, filenameHash)
+	if _, err := os.Stat(path); err != nil {
+		return false
+	}
+	return true
+}
+
+// GetStoredCredentials retrieves collector credentials stored in local file system and then decrypts it
+// using hashed collector name as passphrase.
+func (cr credsGetter) GetStoredCredentials() (api.OpenRegisterResponsePayload, error) {
+	filenameHash, err := hash(cr.conf.CollectorName)
+	if err != nil {
+		return api.OpenRegisterResponsePayload{}, err
+	}
+
+	path := path.Join(cr.conf.CollectorCredentialsPath, filenameHash)
+	creds, err := os.Open(path)
+	if err != nil {
+		return api.OpenRegisterResponsePayload{}, err
+	}
+
+	defer creds.Close()
+	encryptedCreds, err := ioutil.ReadAll(creds)
+	if err != nil {
+		return api.OpenRegisterResponsePayload{}, err
+	}
+
+	collectorCreds, err := decrypt(encryptedCreds, cr.conf.CollectorName)
+	if err != nil {
+		return api.OpenRegisterResponsePayload{}, err
+	}
+
+	var credentialsInfo api.OpenRegisterResponsePayload
+	if err = json.Unmarshal(collectorCreds, &credentialsInfo); err != nil {
+		return api.OpenRegisterResponsePayload{}, err
+	}
+
+	return credentialsInfo, nil
+}
+
+// RegisterCollector registers new collector using registration API and returns collector credentials.
+func (cr credsGetter) RegisterCollector(ctx context.Context) (api.OpenRegisterResponsePayload, error) {
+	baseUrl := strings.TrimSuffix(cr.conf.ApiBaseUrl, "/")
+	u, err := url.Parse(baseUrl)
+	if err != nil {
+		return api.OpenRegisterResponsePayload{}, err
+	}
+	u.Path = registerUrl
+
+	// TODO: just plain hostname or we want to add some custom logic when setting
+	// hostname in request?
+	hostname, err := os.Hostname()
+	if err != nil {
+		return api.OpenRegisterResponsePayload{}, fmt.Errorf("cannot get hostname: %w", err)
+	}
+
+	var buff bytes.Buffer
+	if err = json.NewEncoder(&buff).Encode(api.OpenRegisterRequestPayload{
+		CollectorName: cr.conf.CollectorName,
+		Description:   cr.conf.CollectorDescription,
+		Category:      cr.conf.CollectorCategory,
+		Ephemeral:     cr.conf.Ephemeral,
+		Clobber:       cr.conf.Clobber,
+		Hostname:      hostname,
```

Comment:
Sync with main registration fields please :) 

```suggestion
		CollectorName: se.conf.CollectorName,
		Description:   se.conf.CollectorDescription,
		Category:      se.conf.CollectorCategory,
		Hostname:      hostname,
		Ephemeral:     se.conf.Ephemeral,
		Clobber:       se.conf.Clobber,
		TimeZone:      se.conf.TimeZone,
```